src/manager/evm/conversion.rs

```diff
@@ -0,0 +1,110 @@
+            children: vec![],
+
+            // update these two parameters from caller
+            prev_hash: [0; 32],
+            proof: ethers::core::types::Bytes::default(),
+        };
+        Ok(b)
+    }
+}
+
+impl TryFrom<CrossMsg> for crate::manager::evm::subnet_contract::CrossMsg {
+    type Error = anyhow::Error;
+
+    fn try_from(value: CrossMsg) -> Result<Self, Self::Error> {
+        let c = crate::manager::evm::subnet_contract::CrossMsg {
+            wrapped: value.wrapped,
+            message: crate::manager::evm::subnet_contract::StorableMsg::try_from(value.msg)?,
+        };
+        Ok(c)
+    }
+}
+
+impl TryFrom<IPCAddress> for crate::manager::evm::subnet_contract::Ipcaddress {
+    type Error = anyhow::Error;
+
+    fn try_from(value: IPCAddress) -> Result<Self, Self::Error> {
+        Ok(crate::manager::evm::subnet_contract::Ipcaddress {
+            subnet_id: crate::manager::evm::subnet_contract::SubnetID::try_from(&value.subnet()?)?,
+            raw_address: payload_to_evm_address(value.raw_addr()?.payload())?,
+        })
+    }
+}
+
+impl TryFrom<StorableMsg> for crate::manager::evm::subnet_contract::StorableMsg {
+    type Error = anyhow::Error;
+
+    fn try_from(value: StorableMsg) -> Result<Self, Self::Error> {
+        let c = crate::manager::evm::subnet_contract::StorableMsg {
+            from: crate::manager::evm::subnet_contract::Ipcaddress::try_from(value.from)?,
+            to: crate::manager::evm::subnet_contract::Ipcaddress::try_from(value.to)?,
+            value: ethers::core::types::U256::from_str(&value.value.atto().to_string())?,
+            nonce: value.nonce,
+            // TODO: we might need a more gerneral method to handle the method
+            method: (value.method as u32).to_be_bytes(),
+            params: ethers::core::types::Bytes::from(value.params.to_vec()),
+        };
+        Ok(c)
+    }
+}
+
+impl TryFrom<ChildCheck> for crate::manager::evm::subnet_contract::ChildCheck {
+    type Error = anyhow::Error;
+
+    fn try_from(value: ChildCheck) -> Result<Self, Self::Error> {
+        let c = crate::manager::evm::subnet_contract::ChildCheck {
+            source: crate::manager::evm::subnet_contract::SubnetID::try_from(&value.source)?,
+            checks: value
+                .checks
+                .iter()
+                .map(|c| {
+                    let mut v = [0; 32];
+                    // TODO: we should update the solidity contract to use bytes
+                    v.copy_from_slice(&c.cid().to_bytes()[0..32]);
+                    v
+                })
+                .collect(),
+        };
+        Ok(c)
+    }
+}
+
+impl TryFrom<&SubnetID> for crate::manager::evm::subnet_contract::SubnetID {
+    type Error = anyhow::Error;
+
+    fn try_from(subnet: &SubnetID) -> Result<Self, Self::Error> {
+        Ok(crate::manager::evm::subnet_contract::SubnetID {
+            root: subnet.root_id(),
+            route: agent_subnet_to_evm_addresses(subnet)?,
```

Comment:
Can you remind me what `agent_subnet_to_evm_addresses` does? It is translating from an f4 to an 0x address? What if any of the addresses in the root is not an f4 and it can't be translated in an 0x address?